src/main/java/com/pinterest/secor/util/orc/VectorColumnFiller.java

```diff
@@ -162,6 +165,38 @@ public void convert(JsonElement value, ColumnVector vect, int row) {
         }
     }
 
+    static class MapColumnConverter implements JsonConverter {
+        private JsonConverter[] childrenConverters;
+
+        public MapColumnConverter(TypeDescription schema) {
+            List<TypeDescription> kids = schema.getChildren();
+            childrenConverters = new JsonConverter[kids.size()];
+            for (int c = 0; c < childrenConverters.length; ++c) {
+                childrenConverters[c] = createConverter(kids.get(c));
+            }
+        }
+
+        public void convert(JsonElement value, ColumnVector vect, int row) {
+            if (value == null || value.isJsonNull()) {
+                vect.noNulls = false;
+                vect.isNull[row] = true;
+            } else {
+                MapColumnVector vector = (MapColumnVector) vect;
+                JsonObject obj = value.getAsJsonObject();
+                Set<String> keySet = obj.keySet();
+                vector.lengths[row] = obj.size();
+                vector.offsets[row] = row > 0 ? vector.offsets[row - 1] + vector.lengths[row - 1] : 0;
+                vector.keys.ensureSize(keySet.size(), true);
```

Comment:
Waiting for your investigation.

Reply:
Figured out what the problem was, wrote code to deal with it and also wrote some test code: https://github.com/pinterest/secor/pull/982/commits/8098f91c1d3ffe3046a30c25c6c1fe6ee985fee0. It would be great if @HenryCaiHaiying (or anyone) could take a look and give some feedback. Thanks.